holidays/countries/thailand.py

```diff
@@ -37,131 +200,634 @@ def add_holiday(dt, holiday_name):
+            2042,
+            2045,
+            2048,
+            2050,
+            2053,
+            2056,
+        }
 
-        # New Year's Eve.
-        add_holiday(date(year, DEC, 31), "New Year's Eve")
+        # While Buddhist Holy Days have been observed since the 1900s
+        #   Due to the calendar changes in 1941 (B.E. 2484) and that
+        #   our array only goes up to B.E. 2600; We'll thus only populate
+        #   the data for 1942-2057 (B.E. 2485-2600).
+        # Sources: หนังสือเวียนกรมการปกครอง กระทรวงมหาดไทย ที่ มท 0310.1/ว4
+        #                          ออกเมื่อ 5 กุมภาพันธ์ พ.ศ.2539
+
+        # Start -> 1st Waxing Day of Month 1, 1942: Nov 19, 1941
+        thai_lun_cal_st_date = date(1941, NOV, 19)
+        thai_lun_cal_st_year = 1942
+
+        # Getting the start date of that particular Thai Lunar Calendar Year
+        if 1942 <= year <= 2057:
+            while thai_lun_cal_st_year < year:
+                if thai_lun_cal_st_year in athikamat_years_gregorian:
+                    thai_lun_cal_st_date += td(days=+384)
+                elif thai_lun_cal_st_year in athikawan_years_gregorian:
+                    thai_lun_cal_st_date += td(days=+355)
+                else:
+                    thai_lun_cal_st_date += td(days=+354)
+                thai_lun_cal_st_year += 1
+
+            # !!! Makha Bucha !!!
+            # วันมาฆบูชา
+            # Athikamat: 15th Waxing Day of Month 4
+            #            or 29[1] + 30[2] + 29[3] + 15[4] -1 = 102
+            # Athikawan: 15th Waxing Day of Month 3
+            #            or 29[1] + 30[2] + 15[3] -1 = 73
+            # Pakatimat: 15th Waxing Day of Month 3
+            #            or 29[1] + 30[2] + 15[3] -1 = 73
+            # Status: In-Use
+            makha_bucha_en = "Makha Bucha"
+
+            if year in athikamat_years_gregorian:
+                add_holiday(
+                    thai_lun_cal_st_date + td(days=+102),
+                    makha_bucha_en,
+                )
+            else:
+                add_holiday(
+                    thai_lun_cal_st_date + td(days=+73),
+                    makha_bucha_en,
+                )
+
+            # !!! Wisakha Bucha !!!
+            # วันวิสาขบูชา
+            # Athikamat: 15th Waxing Day of Month 6
+            #            or 177[1-6] + 15[7] -1 = 191
+            # Athikawan: 15th Waxing Day of Month 6
+            #            or 147[1-5] + 15[6] -1 = 161
+            # Pakatimat: 15th Waxing Day of Month 6
+            #            or 147[1-5] + 15[6] -1 = 161
+            # ** Should the date be in April, an extra 30 days is added
+            # Status: In-Use
+            visakha_bucha_en = "Visakha Bucha"
+
+            if year in athikamat_years_gregorian:
+                add_holiday(
+                    thai_lun_cal_st_date + td(days=+191),
+                    visakha_bucha_en,
+                )
+            elif (thai_lun_cal_st_date + td(days=+102)).month == 4:
```

Comment:
```suggestion
            elif (thai_lun_cal_st_date + td(days=+161)).month == 4:
```
I think, it should be 161 here. And this condition never occurs in 1942-2057, so we can't cover it by test :(

Reply:
@PPsyrius, I tried to make full tests for your great work, look at PR in your repo ;)

Reply:
Oof, I know I messed up something when copying off the date calculation from my other file :(

And yeah, that kinda explains why my code doesn't work for specific years - I don't think the `** Should the date be in April, an extra 30 days is added` is needed anyway from another check